src/rendergraph/common/rendergraph/attributeset.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "backend/baseattributeset.h"
+#include "rendergraph/attributeinit.h"
+
+namespace rendergraph {
+class AttributeSet;
+}
+
+class rendergraph::AttributeSet : public rendergraph::BaseAttributeSet {
+  public:
+    AttributeSet(std::initializer_list<AttributeInit> list, const std::vector<QString>& names);
+};
+
+namespace rendergraph {
+template<typename... T>
+AttributeSet makeAttributeSet(const std::vector<QString>& names) {
+    return AttributeSet({(AttributeInit::create<T>())...}, names);
```

Comment:
you can alternatively consider adding an attribute for an array with literals and add some compile time checks then:
```suggestion
template<typename... Ts, int N>
AttributeSet makeAttributeSet(const QString (&names)[N)) {
    static_assert(sizeof...(Ts) == N);
    return AttributeSet({(AttributeInit::create<Ts>())...}, std::vector<QString>(std::cbegin(names), std::cend(names));
```
This can then be used
```cpp
auto set = makeAttributeSet<Foo, Bar>({u"foo"_s, u"bar"_s});
```

If you want to omit the `{}`, you can also use a variadic function template:
```suggestion
template<typename... Ts, typename... Names>
AttributeSet makeAttributeSet(Names... names) {
    static_assert((... && std::is_same_v<Names, QString>)); // extra () is required so the fold-expression gets lexed correctly.
    static_assert(sizeof...(Ts) == sizeof...(Names));
    return AttributeSet({(AttributeInit::create<Ts>())...}, {std::move(names), ...});
```

Both of them still allow the actual strings to be passed at runtime (which is not even needed afaict). I honestly don't care about what version you go for here, I just wanted to nerd out a little ;)

Reply:
Nice! I’ll have a look and pick one, if it plays indeed nice with both the opengl and the scenegraph backend.

Reply:
I couldn't get the version with the variadic function template to compile. Anyway, I prefer sticking with the { }, because the uniforms use that too and I don't want to edit all code.

Also, a comment like:

// extra () is required so the fold-expression gets lexed correctly.

makes me think that this is getting needlessly complicated! 😅

Reply:
Please re-review and resolve conversation.

Reply:
Thanks. added a new (tangentially related comment). 

> makes me think that this is getting needlessly complicated! 😅

Yeah, sorta. I think the compile time arity-checking is very nice, but the difference between the two suggestions is just personal taste IMO.